### ISSUE TITLE
Support device specific vulkan functions in Rust Vulkan generator

### ISSUE THREAD

**Issue by Qwinci** [Dav1dde/glad#410]
Have support for getting global/per-instance/per-device function pointers in Rust Vulkan generator, eg. instead of a single `glad_vulkan::vk::load` have specific ones for the per-instance and per-device function loading like C has `gladLoaderLoadVulkan` which can take an instance and a device.

**Comment by Dav1dde** [comment]
Can you explain why the current signature is not enough? The `load` function takes an `FnMut` which should allow you to capture a device/instance and manipulate it if necessary.

**Comment by Qwinci** [comment]
Yeah I guess the current `load` function works fine, maybe there should be some kind of function which you can use to check if a specific function is device specific so you could use `vkGetDeviceProcAddr` instead of `vkGetInstanceProcAddr` to get the device specific functions for those if you have a specific device?

**Comment by Dav1dde** [comment]
Agreed that would be useful, since this is somewhat tricky ... [due to some driver bugs ...](https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/issues/2323)
